### PR TITLE
added cases for special character search

### PR DESCRIPTION
**Pull request by evr3n**

### CODE REVIEW

File: Tests/Source/Model/TextSearchQueryTests.swift [hunk 1]
```diff
@@ -322,6 +322,65 @@ class TextSearchQueryTests: BaseZMClientMessageTests {
...
+        verifyThatItFindsMessage(withText: "https://www.link.com/something-to-read?q=12&second#reader", whenSearchingFor: "something to read")
+        verifyThatItFindsMessage(withText: "<8000 x a's>", whenSearchingFor: "<8000 x a's>")
+        verifyThatItFindsMessage(withText: "bb бб bb", whenSearchingFor: "бб")
+        verifyThatItFindsMessage(withText: "bb бб bb", whenSearchingFor: "bb")
+        verifyThatItFindsMessage(withText: "苹果", whenSearchingFor: "Ping")
+
+
+
+
+
```
Comment (mikeger) [inline Tests/Source/Model/TextSearchQueryTests.swift]: Looks good. 
```










```
The newlines seems unnecessary.

File: Tests/Source/Model/TextSearchQueryTests.swift [hunk 2]
```diff
@@ -322,6 +322,65 @@ class TextSearchQueryTests: BaseZMClientMessageTests {
...
+        verifyThatItFindsMessage(withText: "aabb", whenSearchingFor: "bb")
+        verifyThatItFindsMessage(withText: "bb aa", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa bb", whenSearchingFor: "aa bb")
+        verifyThatItFindsMessage(withText: "aa", whenSearchingFor: "bb ")
+        verifyThatItFindsMessage(withText: "aabb", whenSearchingFor: "aa\nbb")
+        verifyThatItFindsMessage(withText: "aa aa aa", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa bb aa", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa bb", whenSearchingFor: "aa cc")
+        verifyThatItFindsMessage(withText: "aa aa aa", whenSearchingFor: "aa aa")
+        verifyThatItFindsMessage(withText: "aabb", whenSearchingFor: "a")
```
Comment (mikeger) [inline Tests/Source/Model/TextSearchQueryTests.swift]: I thought we don't have any support for searching one letter. Should there be the cases for it?

Comment (daehn) [inline Tests/Source/Model/TextSearchQueryTests.swift]: Creating the `TextSearchQuery` object should fail in that case. I will check the test setup if it handles this correctly.

File: Tests/Source/Model/TextSearchQueryTests.swift [hunk 3]
```diff
@@ -322,6 +322,65 @@ class TextSearchQueryTests: BaseZMClientMessageTests {
         verifyThatItFindsMessage(withText: "aa aa aa", whenSearchingFor: "aa")
         verifyThatItFindsMessage(withText: "aa.aa", whenSearchingFor: "aa")
         verifyThatItFindsMessage(withText: "11:45", whenSearchingFor: "11:45")
+        verifyThatItFindsMessage(withText: "aabb", whenSearchingFor: "aabb")
+        verifyThatItFindsMessage(withText: "aabb", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aabb", whenSearchingFor: "bb")
+        verifyThatItFindsMessage(withText: "bb aa", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa bb", whenSearchingFor: "aa bb")
+        verifyThatItFindsMessage(withText: "aa", whenSearchingFor: "bb ")
```
Comment (daehn) [inline Tests/Source/Model/TextSearchQueryTests.swift]: Does this test pass? Searching for `"bb "` should not return a message with text `"aa"`

File: Tests/Source/Model/TextSearchQueryTests.swift [hunk 4]
```diff
@@ -322,6 +322,65 @@ class TextSearchQueryTests: BaseZMClientMessageTests {
...
+        verifyThatItFindsMessage(withText: "aabb", whenSearchingFor: "aa\nbb")
+        verifyThatItFindsMessage(withText: "aa aa aa", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa bb aa", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa bb", whenSearchingFor: "aa cc")
```
Comment (daehn) [inline Tests/Source/Model/TextSearchQueryTests.swift]: This one should not pass as well.

File: Tests/Source/Model/TextSearchQueryTests.swift [hunk 5]
```diff
@@ -322,6 +322,65 @@ class TextSearchQueryTests: BaseZMClientMessageTests {
...
+        verifyThatItFindsMessage(withText: "aa-bb", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa/bb", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa/bb", whenSearchingFor: "bb")
+        verifyThatItFindsMessage(withText: "aa:bb", whenSearchingFor: "aa")
+        verifyThatItFindsMessage(withText: "aa:bb", whenSearchingFor: "bb")
+        verifyThatItFindsMessage(withText: "@peter", whenSearchingFor: "peter")
+        verifyThatItFindsMessage(withText: "rené", whenSearchingFor: "Rene")
+        verifyThatItFindsMessage(withText: "https://www.link.com/something-to-read?q=12&second#reader", whenSearchingFor: "something to read")
+        verifyThatItFindsMessage(withText: "<8000 x a's>", whenSearchingFor: "<8000 x a's>")
+        verifyThatItFindsMessage(withText: "bb бб bb", whenSearchingFor: "бб")
```
Comment (mikeger) [inline Tests/Source/Model/TextSearchQueryTests.swift]: This test case seems not correct, I guess the aim is to proof that searching for cyrillic `b`, i.e. `б` in the text with latin b is working. It should be "bb" matching search for "бб". Same applies for the next test, but reversed, i.e. searching for Cyrillic B should return the latin one.

**Comment by mikeger** [comment]
@evr3n did you run the tests?

File: Tests/Source/Model/TextSearchQueryTests.swift (hunk 4 above)
Comment (evr3n) [inline Tests/Source/Model/TextSearchQueryTests.swift]: Is there a way to verify that it doesn't find? To e.g. test the one letter cases

Comment (daehn) [inline Tests/Source/Model/TextSearchQueryTests.swift]: Yes there is like this:
```swift 
verifyThatItFindsMessage(withText: "aabb", whenSearchingFor: "a", shouldFind: false)
```

File: Tests/Source/Model/TextSearchQueryTests.swift (hunk 2 above)
Comment (daehn) [inline Tests/Source/Model/TextSearchQueryTests.swift]: @evr3n I pushed some changes on this branch to ensure the tests are not crashing anymore but failing instead if the creation of a `TextSearchQuery` fails (which happens when the search query is shorter than 2 characters long).

**Comment by mikeger** [comment]
@evr3n this works now, tests are ✅ ?

**Comment by daehn** [comment]
@mikeger test were run by CircleCi and are

**Pull request merged**